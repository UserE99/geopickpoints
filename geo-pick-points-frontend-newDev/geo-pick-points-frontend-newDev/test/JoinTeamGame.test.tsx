import { joinInTeam, unjoinTeam, TeamData, UnjoinTeamData } from "../src/actions/JoinTeamGame";
import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import fetchMock from 'jest-fetch-mock';

// Initialisiere fetchMock
fetchMock.enableMocks();

// Testdaten
const teamData: TeamData = {
    nickName: "Team A",
    playerID: "123",
    teamId: "T001"
};

const unjoinData: UnjoinTeamData = {
    teamId: "T001",
    playerID: "123"
};

// Mock fÃ¼r den Store
const createTestStore = () => {
    return configureStore({
        reducer: {

        },

    });
};

describe("joinInTeam Thunk", () => {
    beforeEach(() => {
        fetchMock.resetMocks();
    });

    it("should successfully join a team", async () => {
        // Mock einer erfolgreichen Antwort
        fetchMock.mockResponseOnce(
            JSON.stringify({ success: true, data: teamData }),
            { status: 200 }
        );

        const store = createTestStore();

        // Aufruf des Thunks
        const result = await store.dispatch(joinInTeam(teamData));

        expect(result.type).toBe("joinInTeam/fulfilled");
        expect(result.payload).toEqual({ success: true, data: teamData });

        // Snapshot testen
        expect(result).toMatchSnapshot();
    });

    it("should handle error when joining team", async () => {
        // Mock einer fehlgeschlagenen Antwort
        fetchMock.mockResponseOnce(
            JSON.stringify({ message: "Failed to join team" }),
            { status: 400 }
        );

        const store = createTestStore();

        // Aufruf des Thunks
        const result = await store.dispatch(joinInTeam(teamData));

        expect(result.type).toBe("joinInTeam/rejected");
        expect(result.error.message).toBe("Rejected");

        // Snapshot testen
        expect(result).toMatchSnapshot();
    });
});

describe("unjoinTeam Thunk", () => {
    beforeEach(() => {
        fetchMock.resetMocks();
    });

    it("should successfully unjoin a team", async () => {
        // Mock einer erfolgreichen Antwort
        fetchMock.mockResponseOnce(
            JSON.stringify({ success: true }),
            { status: 200 }
        );

        const store = createTestStore();

        // Aufruf des Thunks
        const result = await store.dispatch(unjoinTeam(unjoinData));

        expect(result.type).toBe("unjoinTeam/fulfilled");
        expect(result.payload).toEqual({ success: true });

        // Snapshot testen
        expect(result).toMatchSnapshot();
    });

    it("should handle error when unjoining team", async () => {
        // Mock einer fehlgeschlagenen Antwort
        fetchMock.mockResponseOnce(
            JSON.stringify({ message: "Failed to unjoin team" }),
            { status: 400 }
        );

        const store = createTestStore();

        // Aufruf des Thunks
        const result = await store.dispatch(unjoinTeam(unjoinData));

        expect(result.type).toBe("unjoinTeam/rejected");
        expect(result.error.message).toBe("Rejected");

        // Snapshot testen
        expect(result).toMatchSnapshot();
    });
});

