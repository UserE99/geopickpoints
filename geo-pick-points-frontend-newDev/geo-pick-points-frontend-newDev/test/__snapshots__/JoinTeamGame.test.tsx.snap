// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinInTeam Thunk should handle error when joining team 1`] = `
{
  "error": {
    "message": "Rejected",
  },
  "meta": {
    "aborted": false,
    "arg": {
      "nickName": "Team A",
      "playerID": "123",
      "teamId": "T001",
    },
    "condition": false,
    "rejectedWithValue": true,
    "requestId": "tBrKxnG9ULwUhiin9UC1F",
    "requestStatus": "rejected",
  },
  "payload": "HTTP error! Status: 400",
  "type": "joinInTeam/rejected",
}
`;

exports[`joinInTeam Thunk should successfully join a team 1`] = `
{
  "meta": {
    "arg": {
      "nickName": "Team A",
      "playerID": "123",
      "teamId": "T001",
    },
    "requestId": "qk0oUwmgAsV3OE02z1-WD",
    "requestStatus": "fulfilled",
  },
  "payload": {
    "data": {
      "nickName": "Team A",
      "playerID": "123",
      "teamId": "T001",
    },
    "success": true,
  },
  "type": "joinInTeam/fulfilled",
}
`;

exports[`unjoinTeam Thunk should handle error when unjoining team 1`] = `
{
  "error": {
    "message": "Rejected",
  },
  "meta": {
    "aborted": false,
    "arg": {
      "playerID": "123",
      "teamId": "T001",
    },
    "condition": false,
    "rejectedWithValue": true,
    "requestId": "jeu8nU0TmLnMBvyoBXGhz",
    "requestStatus": "rejected",
  },
  "payload": "HTTP error! Status: 400",
  "type": "unjoinTeam/rejected",
}
`;

exports[`unjoinTeam Thunk should successfully unjoin a team 1`] = `
{
  "meta": {
    "arg": {
      "playerID": "123",
      "teamId": "T001",
    },
    "requestId": "nkUpmfTXyufUnw5G-Vj_g",
    "requestStatus": "fulfilled",
  },
  "payload": {
    "success": true,
  },
  "type": "unjoinTeam/fulfilled",
}
`;
